Merging Develop to Main

KazeemHamzat opened the pull request:
Merging Develop to main.

SheenaSharma committed:
Create CodeSystem-England-MessageEventsBars.xml

Changes for url in Codesystem File.

SheenaSharma committed:
Create CodeSystem-England-MessageReasonBars.xml

Change in URL for Codesystem file

SheenaSharma committed:
Change in the description and file names.

Change in the description and file names.

Commit:
Change capitalisation and remove CS from base dir

SheenaSharma committed:
Changed from Bars to BARS.

Changed from Bars to BARS.

SheenaSharma committed:
Changed indentation from 4 spaces to 2

Changed indentation from 4 spaces to 2

SheenaSharma committed:
Merge 12697177164c708ab573dec3de98785efb28ab8e into 448e4ec41c878650f40577cecccd4a9d96b0fff5

ryma2fhir committed:
GitHub Action: Find and Replace

ryma2fhir committed:
Merge pull request #441 from NHSDigital/Feature-IOPS-2581-Add-BARS-CodeSystem-message-event-bars-to-NHSE-IG

Feature iops 2581 add bars code system message event bars to nhse ig

digitalhealthmartin committed:
Add CodeSystem England-HTTPErrorCode (#440)

* Add CodeSystem England-HTTPErrorCode

* Updated indentation and file name

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

digitalhealthmartin committed:
Add Extension LocationExtension (#442)

* Add LocationExtension

* Update LocationExtension URL, version, description and purpose

* Added missing contact, publisher and copyright to LocationExtension

* Update fixedUri in LocationExtension

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

KazeemHamzat committed:
Update Extension-England-TypedPeriod (#463)

* Update Extension-England-TypedPeriod

Updated Extension-England-TypedPeriodExtension-England-TypedPeriod, CodeSystem-England-PeriodType  & ValueSet-England-PeriodType

* Update Extension-England-TypedPeriod

Extension-England-TypedPeriod

* Update CodeSystem-England-PeriodType

Update CodeSystem-England-PeriodType

* Update CodeSystem-England-PeriodType

Update CodeSystem-England-PeriodType

* Create-CodeSystem & ValueSet -CallbackConsent

Create-CodeSystem & ValueSet -CallbackConsent

* Rename ValueSet

* Update code display to Consent to Callback

Update code display to Consent to Callback

* Update CodeSystem-England-PeriodType.

Update CodeSystem-England-PeriodType

* Renaming Assets

1. CodeSystem-England-ConsentType.xml
2. ValueSet-England-Consent-Type.xml

* Update CodeSystem-England-ConsentType

Update CodeSystem-England-ConsentType

digitalhealthmartin reviewed (commented):
Four files are 'updated' and others are new. Only 'updated' files have upped version num - so the other files are new hence 1.0.0? Also the 'new' files are status 'draft', is this right or should be 'active' as now going into main branch

ryma2fhir reviewed (dismissed):
as per Martins comment, 1.0.0 means it has been balloted and made active. Until then,  it should be draft and 0.x.x

digitalhealthmartin reviewed (commented):
Don't think this is up to date yet and thus out of date. There were changes in https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/466/files whicn include version num updates which don't seem to reflect here

KazeemHamzat committed:
Changing Version No to 0.1.0 due to DRAFT Status (#466)

* Merge develop into main (#432)

* Change line breaks to LF, replace tab with 2 spaces

* added find and replace workflow

* Find and Replace

* Update findAndReplace.yml

* changed to allow signed commits

* GitHub Action: Find and Replace

* Added England version of UKCore-TypedPeriod, extended OrgRole with it

* Update findAndReplace.yml

Change to allow for GitHub actions to rerun after merge from this action. If not allowed, then PR gets hung-up waiting for checks to be done without actually running them.

* Resolution of Sprint 1 Consolidation Call (#420)

* Update Status of CP-IS & FGM Assets

1. Change the status of CP-IS & FGM assets to active.
2. Change the published date to 2023-11-09 when they are due to be published.

* Update ValueSet-England-ChildProtectionPlan.xml

Update ValueSet-England-ChildProtectionPlan to 2 space tab

* Update ValueSet-England-ChildProtectionPlan.xml

Update ValueSet-England-ChildProtectionPlan to 2 spaces tab

* Delete .github/workflows/FindReplaceTabSpace.yml

Superseded by other find and replace ation

* Remove primary here rather than separate pr

* GitHub Action: Find and Replace

* Re add min 1 to period in Org role

* Review tweaks

* Remove another xml header tag

* Create PFS CodeSystem as an NHS England IG Asset (#427)

* Create PFS CodeSystem as an NHS England IG Asset

* GitHub Action: Find and Replace

* IOPS2261 name, date and description updated

Rename of file, id, url, name and title.
date changed to 2023-11-20
description value changed to suggested text.

---------

Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Extension sub url, and wording changes, valueset url include change

* Sprint2 ord/ods naming systems + not-present codesystems (#424)

* NamingSystems and basic not-present CodeSystems

* Switch OrgRole to new codesystem url

* Fix duff id/name

* Remove spaces from name

* Remove spaces from name

* Set naming systems to draft

* GitHub Action: Find and Replace

* Remove namingsystems, tweak codesystems to have ODS contact details, and url for the codes

* GitHub Action: Find and Replace

* Tweaks following review

* GitHub Action: Find and Replace

* reword

* GitHub Action: Find and Replace

---------

Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Re-add workflow

* Asset changes from Sprint 2 consolidation call (#431)

* Asset changes from Sprint 2 consolidation call

* Several updates to assets

* Updated ValueSet TypedDateTime

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Remove ORD duplicate assets

---------

Co-authored-by: Ryan May <ryan.may2@nhs.net>
Co-authored-by: ryma2fhir <ryma2fhir@users.noreply.github.com>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Adam Page <adam.page2@nhs.net>
Co-authored-by: KazeemHamzat <113706558+KazeemHamzat@users.noreply.github.com>
Co-authored-by: Adam Page <115618858+APageNHS@users.noreply.github.com>
Co-authored-by: adrianhand2 <103494146+adrianhand2@users.noreply.github.com>
Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Merge develop into main (#434)

* Change line breaks to LF, replace tab with 2 spaces

* added find and replace workflow

* Find and Replace

* Update findAndReplace.yml

* changed to allow signed commits

* GitHub Action: Find and Replace

* Added England version of UKCore-TypedPeriod, extended OrgRole with it

* Update findAndReplace.yml

Change to allow for GitHub actions to rerun after merge from this action. If not allowed, then PR gets hung-up waiting for checks to be done without actually running them.

* Resolution of Sprint 1 Consolidation Call (#420)

* Update Status of CP-IS & FGM Assets

1. Change the status of CP-IS & FGM assets to active.
2. Change the published date to 2023-11-09 when they are due to be published.

* Update ValueSet-England-ChildProtectionPlan.xml

Update ValueSet-England-ChildProtectionPlan to 2 space tab

* Update ValueSet-England-ChildProtectionPlan.xml

Update ValueSet-England-ChildProtectionPlan to 2 spaces tab

* Delete .github/workflows/FindReplaceTabSpace.yml

Superseded by other find and replace ation

* Remove primary here rather than separate pr

* GitHub Action: Find and Replace

* Re add min 1 to period in Org role

* Review tweaks

* Remove another xml header tag

* Create PFS CodeSystem as an NHS England IG Asset (#427)

* Create PFS CodeSystem as an NHS England IG Asset

* GitHub Action: Find and Replace

* IOPS2261 name, date and description updated

Rename of file, id, url, name and title.
date changed to 2023-11-20
description value changed to suggested text.

---------

Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Extension sub url, and wording changes, valueset url include change

* Sprint2 ord/ods naming systems + not-present codesystems (#424)

* NamingSystems and basic not-present CodeSystems

* Switch OrgRole to new codesystem url

* Fix duff id/name

* Remove spaces from name

* Remove spaces from name

* Set naming systems to draft

* GitHub Action: Find and Replace

* Remove namingsystems, tweak codesystems to have ODS contact details, and url for the codes

* GitHub Action: Find and Replace

* Tweaks following review

* GitHub Action: Find and Replace

* reword

* GitHub Action: Find and Replace

---------

Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Re-add workflow

* Asset changes from Sprint 2 consolidation call (#431)

* Asset changes from Sprint 2 consolidation call

* Several updates to assets

* Updated ValueSet TypedDateTime

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Remove ORD duplicate assets

* Add missing file (#433)

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

---------

Co-authored-by: Ryan May <ryan.may2@nhs.net>
Co-authored-by: ryma2fhir <ryma2fhir@users.noreply.github.com>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Adam Page <adam.page2@nhs.net>
Co-authored-by: KazeemHamzat <113706558+KazeemHamzat@users.noreply.github.com>
Co-authored-by: Adam Page <115618858+APageNHS@users.noreply.github.com>
Co-authored-by: adrianhand2 <103494146+adrianhand2@users.noreply.github.com>
Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Merge develop into main (#436)

* Change line breaks to LF, replace tab with 2 spaces

* added find and replace workflow

* Find and Replace

* Update findAndReplace.yml

* changed to allow signed commits

* GitHub Action: Find and Replace

* Added England version of UKCore-TypedPeriod, extended OrgRole with it

* Update findAndReplace.yml

Change to allow for GitHub actions to rerun after merge from this action. If not allowed, then PR gets hung-up waiting for checks to be done without actually running them.

* Resolution of Sprint 1 Consolidation Call (#420)

* Update Status of CP-IS & FGM Assets

1. Change the status of CP-IS & FGM assets to active.
2. Change the published date to 2023-11-09 when they are due to be published.

* Update ValueSet-England-ChildProtectionPlan.xml

Update ValueSet-England-ChildProtectionPlan to 2 space tab

* Update ValueSet-England-ChildProtectionPlan.xml

Update ValueSet-England-ChildProtectionPlan to 2 spaces tab

* Delete .github/workflows/FindReplaceTabSpace.yml

Superseded by other find and replace ation

* Remove primary here rather than separate pr

* GitHub Action: Find and Replace

* Re add min 1 to period in Org role

* Review tweaks

* Remove another xml header tag

* Create PFS CodeSystem as an NHS England IG Asset (#427)

* Create PFS CodeSystem as an NHS England IG Asset

* GitHub Action: Find and Replace

* IOPS2261 name, date and description updated

Rename of file, id, url, name and title.
date changed to 2023-11-20
description value changed to suggested text.

---------

Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Extension sub url, and wording changes, valueset url include change

* Sprint2 ord/ods naming systems + not-present codesystems (#424)

* NamingSystems and basic not-present CodeSystems

* Switch OrgRole to new codesystem url

* Fix duff id/name

* Remove spaces from name

* Remove spaces from name

* Set naming systems to draft

* GitHub Action: Find and Replace

* Remove namingsystems, tweak codesystems to have ODS contact details, and url for the codes

* GitHub Action: Find and Replace

* Tweaks following review

* GitHub Action: Find and Replace

* reword

* GitHub Action: Find and Replace

---------

Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Re-add workflow

* Asset changes from Sprint 2 consolidation call (#431)

* Asset changes from Sprint 2 consolidation call

* Several updates to assets

* Updated ValueSet TypedDateTime

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Remove ORD duplicate assets

* Add missing file (#433)

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Fix capitalisation for code values (#435)

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

---------

Co-authored-by: Ryan May <ryan.may2@nhs.net>
Co-authored-by: ryma2fhir <ryma2fhir@users.noreply.github.com>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Adam Page <adam.page2@nhs.net>
Co-authored-by: KazeemHamzat <113706558+KazeemHamzat@users.noreply.github.com>
Co-authored-by: Adam Page <115618858+APageNHS@users.noreply.github.com>
Co-authored-by: adrianhand2 <103494146+adrianhand2@users.noreply.github.com>
Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Merge develop into main (#444)

* Create CodeSystem-England-MessageEventsBars.xml

Changes for url in Codesystem File.

* Create CodeSystem-England-MessageReasonBars.xml

Change in URL for Codesystem file

* Change in the description and file names.

Change in the description and file names.

* Change capitalisation and remove CS from base dir

* Changed from Bars to BARS.

Changed from Bars to BARS.

* Changed indentation from 4 spaces to 2

Changed indentation from 4 spaces to 2

* GitHub Action: Find and Replace

* Add CodeSystem England-HTTPErrorCode (#440)

* Add CodeSystem England-HTTPErrorCode

* Updated indentation and file name

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

* Add Extension LocationExtension (#442)

* Add LocationExtension

* Update LocationExtension URL, version, description and purpose

* Added missing contact, publisher and copyright to LocationExtension

* Update fixedUri in LocationExtension

---------

Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>

---------

Co-authored-by: Sheena Sharma <sheena.sharma5@nhs.net>
Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>
Co-authored-by: Sheena Sharma <99985081+SheenaSharma@users.noreply.github.com>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>

* Update Extension-England-TypedPeriod

Updated Extension-England-TypedPeriodExtension-England-TypedPeriod, CodeSystem-England-PeriodType  & ValueSet-England-PeriodType

* Update Extension-England-TypedPeriod

Extension-England-TypedPeriod

* Update CodeSystem-England-PeriodType

Update CodeSystem-England-PeriodType

* Update CodeSystem-England-PeriodType

Update CodeSystem-England-PeriodType

* Create-CodeSystem & ValueSet -CallbackConsent

Create-CodeSystem & ValueSet -CallbackConsent

* Rename ValueSet

* Update code display to Consent to Callback

Update code display to Consent to Callback

* link to codesystem oage fixed (#462)

* Update CodeSystem-England-PeriodType.

Update CodeSystem-England-PeriodType

* Renaming Assets

1. CodeSystem-England-ConsentType.xml
2. ValueSet-England-Consent-Type.xml

* Update CodeSystem-England-ConsentType

Update CodeSystem-England-ConsentType

* Changing Version No to 0.1.0 due to DRAFT Status

Changing Version No to 0.1.0 as the assets are still in DRAFT status

---------

Co-authored-by: Martin Fison <82446327+digitalhealthmartin@users.noreply.github.com>
Co-authored-by: Ryan May <ryan.may2@nhs.net>
Co-authored-by: ryma2fhir <ryma2fhir@users.noreply.github.com>
Co-authored-by: Ryan May <112563297+ryma2fhir@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Adam Page <adam.page2@nhs.net>
Co-authored-by: Adam Page <115618858+APageNHS@users.noreply.github.com>
Co-authored-by: adrianhand2 <103494146+adrianhand2@users.noreply.github.com>
Co-authored-by: SheenaSharma <sheena.sharma5@nhs.net>
Co-authored-by: Martin Fison <MAFI8@hscic.gov.uk>
Co-authored-by: Sheena Sharma <99985081+SheenaSharma@users.noreply.github.com>
Co-authored-by: NaminderSoorma <naminder.soorma@nhs.net>

KazeemHamzat commented:
> > Don't think this is up to date yet and thus out of date. There were changes in https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/466/files whicn include version num updates which don't seem to reflect here
> 
> The changes will reflect here once the PR for Feature Branch to Stu1_develop is merged. First need to merge [Change Version No PR to develop](https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/466), then this one to main

The feature branch has been merged into develop, and the expected changes are now reflected in this PR.

KazeemHamzat committed:
Merge branch 'stu1_main' of https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide into stu1_develop

NaminderSoorma reviewed (approved):
The only odd thing i noticed was version changes to lower numbers. E.g. from 1.0.0 to 0.1.0

KazeemHamzat commented:
> The only odd thing i noticed was version changes to lower numbers. E.g. from 1.0.0 to 0.1.0

They were reversed to 0.1.0 as the assets are still in draft status.

vickyjaiswal0308 commented:
I can see the QC checker is failing because of the wrong naming convention - can it be fixed?

KazeemHamzat commented:
> I can see the QC checker is failing because of the wrong naming convention - can it be fixed?

The files affected are not related to my work. They were transferred from another project, and needed to retain their old canonical URL. So they will always fail. QC checker. However, the solution might be to allow them use their old meta rather than aligning their meta to NHSE Design approach. That can be tested in a new feature branch if that is approved by DA.
The affected files:
**Valuesets**
1. nhsdigital-message-events.xml`
**CodeSystems**
1. message-event.xml
2. CodeSystem-England-MessageReasonBARS.xml
3. CodeSystem-England-MessageEventsBARS.xml
4. CodeSystem-England-HTTPErrorCodes.xml

DaveBarnet reviewed (commented):
A couple of the changes (England-MessageReasonBARS & England-HTTPErrorCodes) put the version numbers back (from 1.0.0 to 0.1.0). Is that what is intended?

KazeemHamzat commented:
> A couple of the changes (England-MessageReasonBARS & England-HTTPErrorCodes) put the version numbers back (from 1.0.0 to 0.1.0). Is that what is intended?

Yes. Previous feedback says their version num. should be 0.1.0, as their status is DRAFT.

ryma2fhir commented:
> > A couple of the changes (England-MessageReasonBARS & England-HTTPErrorCodes) put the version numbers back (from 1.0.0 to 0.1.0). Is that what is intended?
> 
> Yes. Previous feedback says their version num. should be 0.1.0, as their status is DRAFT.

Apologies, I meant just the new assets.
All draft should be <1.0.0, but if they are being used already then this may cause confusion, especially if they are then made active and we have 2 versions with 1.0.0. 

@NaminderSoorma  Why do we have BARS assets in the main IG, shouldn't it be in the BARS IG where they can keep ownership?

ryma2fhir commented:
Should the consent assets be in the NHSE IG? I'd have expected it to be in the NHSE Programmes IG as it is programme specific, and not a common asset
